ArchUnitNET/Fluent/ICondition.cs

```diff
@@ -0,0 +1,10 @@
+﻿using ArchUnitNET.Domain;
+
+namespace ArchUnitNET.Fluent
+{
+    public interface ICondition<TRuleType> : IHasFailDescription where TRuleType : ICanBeAnalyzed
+    {
+        bool Check(TRuleType obj, Architecture architecture);
+        bool CheckNull();
```

Comment:
what is the main purpose of it?

Reply:
You can use `Should().(Not)Exist()` to check whether there are objects with given properties in the architecture. When you evaluate the rule and there are no objects with the given properties, `CheckNull()` is called for all conditions. It always returns true except for the `Should().Exist()` condition, which returns false if there are no objects found.